### PR TITLE
[dstorage] port updated for 1.0.2 release

### PR DESCRIPTION
**Pull request by walbourn**
* Updates the port for the latest release of `Microsoft.Direct3D.DirectStorage` per [this blog post](https://devblogs.microsoft.com/directx/directstorage-sdk-1-0-2/).
* Changes for upstream reorganization of package layout

### CODE REVIEW

**Comment by walbourn** [comment]
Verified the supported triplets locally:

```
vcpkg install dstorage:x86-windows
vcpkg install dstorage:x86-windows-static-md
vcpkg install dstorage:x64-windows
vcpkg install dstorage:x64-windows-static-md
vcpkg install dstorage:arm-windows
vcpkg install dstorage:arm64-windows
vcpkg install dstorage:arm64-windows-static-md
```

File: ports/dstorage/vcpkg.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "name": "dstorage",
-  "version": "1.0.0",
-  "port-version": 1,
+  "version": "1.0.2",
   "description": "DirectStorage for Windows",
   "homepage": "https://aka.ms/directstorage/",
+  "documentation": "https://github.com/microsoft/DirectStorage",
   "license": null,
```
Comment (JonLiu1993) [inline ports/dstorage/vcpkg.json:7]:
```suggestion
  "license": "MIT",
```

Comment (JonLiu1993) [reply ports/dstorage/vcpkg.json:7]: @walbourn ,Thanks for your pr, I checked and the license here should be "MIT"?
https://github.com/microsoft/DirectStorage/blob/main/LICENSE

Comment (walbourn) [reply ports/dstorage/vcpkg.json:7]: @JonLiu1993  The samples are MIT, but for the NuGet itself, there's a custom license file which is why I used ``null``.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the PR!

**Pull request merged**